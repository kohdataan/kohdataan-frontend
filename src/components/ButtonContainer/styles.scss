@import '../../styles/colors.scss';

.button-secondary {
  font-family: 'Poppins', sans-serif;
  font-size: 16px;
  padding: 5px 10px;
  font-weight: bold;
  text-align: center;
  text-transform: uppercase;
  background-color: $orange;
  border: solid 1px $orange;
  border-radius: 30px;
    &:hover {
      cursor: pointer;
    }
}

