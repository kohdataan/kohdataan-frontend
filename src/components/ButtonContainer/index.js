import React, { memo } from 'react'
import './styles.scss'
import propTypes from 'prop-types'

const ButtonContainer = props => {
  const { className, children, secondary, onClick, label, role } = props
  return (
    <button
      type="button"
      className={`button-container ${
        secondary ? `button-secondary` : ``
      } ${className}`}
      onClick={onClick}
      onKeyPress={onClick}
      tabIndex="0"
      aria-label={label}
      role={role}
    >
      {children}
    </button>
  )
}

ButtonContainer.propTypes = {
  className: propTypes.string.isRequired,
  children: propTypes.oneOfType([
    propTypes.string,
    propTypes.instanceOf(Array),
    propTypes.node,
  ]).isRequired,
  onClick: propTypes.func.isRequired,
  secondary: propTypes.bool,
  label: propTypes.string,
  role: propTypes.string,
}

ButtonContainer.defaultProps = {
  secondary: false,
  label: '',
  role: 'button',
}

export default memo(ButtonContainer)
