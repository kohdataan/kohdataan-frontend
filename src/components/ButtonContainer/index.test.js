import React from 'react'
import { create } from 'react-test-renderer'
import ButtonContainer from './index'

describe('Button container', () => {
  test('it matches the snapshot', () => {
    const component = create(<ButtonContainer />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
