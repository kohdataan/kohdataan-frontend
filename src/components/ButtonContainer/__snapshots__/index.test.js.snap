// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button container it matches the snapshot 1`] = `
<button
  className="button-container undefined"
  tabIndex="0"
  type="button"
/>
`;
